File: frontend/project/src/services/knowledge-ai-service.ts
import { apiClient } from '../lib/api';
import type { PageResponse } from '../types/api';

export interface KnowledgeAskRequest {
  workspaceId: string;
  departmentId: string;
  projectId?: string;
  question: string;
}

export interface KnowledgeSearchRequest {
  workspaceId: string;
  departmentId: string;
  projectId?: string;
  query: string;
  category?: string;
  documentType?: string;
}

export interface KnowledgeSource {
  id: string;
  title: string;
  type: string;
  category: string;
  summary: string;
  workspaceName: string;
  departmentName: string;
  projectName: string;
  lastUpdated: string;
  version: number;
}

export interface KnowledgeAIResponse {
  answer: string;
  sources: KnowledgeSource[];
  confidence: string;
  missingInformation: string;
  suggestedRelatedDocuments: string[];
  executionTime: number;
  timestamp: string;
}

export function knowledgeAIService() {
  const base = '/knowledge/ai';
  return {
    ask: (data: KnowledgeAskRequest) =>
      apiClient.post<KnowledgeAIResponse>(`${base}/ask`, data),

    search: (data: KnowledgeSearchRequest) =>
      apiClient.post<KnowledgeSource[]>(`${base}/search`, data),

    getHistory: (workspaceId: string, page?: number, size?: number) => {
      const params: Record<string, unknown> = { workspaceId };
      if (page != null) params.page = page;
      if (size != null) params.size = size;
      return apiClient.get<PageResponse<KnowledgeAIResponse>>(`${base}/history`, { params });
    },
  };
}

File: backend/src/main/java/com/trio/backend/service/KnowledgeAIService.java
package com.trio.backend.service;

import com.trio.backend.dto.ai.KnowledgeAIResponse;
import com.trio.backend.dto.ai.KnowledgeSource;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.UUID;

public interface KnowledgeAIService {

    KnowledgeAIResponse ask(UUID workspaceId, UUID departmentId, UUID projectId, String question);

    List<KnowledgeSource> search(UUID workspaceId, UUID departmentId, UUID projectId, String query);

    Page<?> getHistory(UUID workspaceId, int page, int size);
}
